### ISSUE TITLE
Input gestures dialog: remove gesture with delete key

### ISSUE THREAD

**Issue by SaschaCowley** [nvaccess/nvda#16816]
### Is your feature request related to a problem? Please describe.

In the input gestures dialog, the only way to delete a gesture is with the "Remove" button that becomes enabled when a gesture is selected.

### Describe the solution you'd like

In line with standard practice for Windows applications, allow removal of a selected gesture with the delete (`Del`) key on the keyboard.

### Describe alternatives you've considered

The status quo.

### Additional context

This is a standard pattern for Windows application, for which users may have muscle memory.

**Comment by Adriani90** [comment]
This applies for speech dictionaries and the expressions added as well I think.
In my view there should be a confirmation promt when pressing del, asking whether the user really wants to delete the entry or not. This is because it is more probable that people press del accidentally, while pressing the remove button accidentally is much less probable.

**Comment by CyrilleB79** [comment]
Yes, applies for speech dictionaries, but also symbol dialog and profile dialog while at it.

I think that using del is a good UX generally.

I am unsure about the confirmation dialog:
* If we add a confirmation dialog only for `del` keypress, but not for "Delete" button, it would be a somewhat inconsistent UX that may be misleading
* if we add a confirmation dialog in all cases (`del` keypress and "Delete" button), it may be quite heavy, heavier than today.
* if we do not add a confirmation dialog in any case, we should ensure that the speech feedback is clear; this should be OK for the braille since the displayed text will update. If the feedback is clear, there is no problem with this solution since the user can undo the changes dones in this dialog by pressing "Cancel" button.

**Comment by XLTechie** [comment]
I agree with @CyrilleB79. This is a good idea, but no confirmation dialog should be used. That is much overkill since the whole thing can be cancelled before applying.
I think users are generally careful about the del key already, given what it can do.

Windows doesn't request confirmation when deleting files to the recycle bin, and at least in many contexts doesn't give speech feedback either.
If the operation can be cancelled (as it can), this seems analogous, especially if we speak a "gesture deleted" or similar message.

Also, let us not forget, that gestures can be reset.

**Comment by wmhn1872265132** [comment]
Also available for Add-on Store's Installed add-ons and Installed incompatible add-ons tabs

**Comment by SaschaCowley** [comment]
If a confirmation dialog were to be implemented, I would hope it would have a "don't prompt in future" checkbox. Regardless, having deletion confirmation dialogs should probably be a separate issue.

**Comment by amirmahdifard** [comment]
hi everyone partisipated in this issue.
I have currently implemented the delete key exactly as the result of the conversation here.
Do you want also a new shortcut? For example a shortcut key to add a new gesture while focused on the script?
Currently, with my implementation, When you focus a gesture and press delete button, you delete the gesture. Also, I added a context menu that holdes the same as buttons, For eas of access But, If you have any shortcut idea for add a new gesture when focused on a script, I'm happy to implement it while i'm at it. Thanks!

**Comment by CyrilleB79** [comment]
While `Delete` key is quite an obvious shortcut key, there is no obvious key to for the "Add" action. Let's keep the "Add" action as is.
Useres who want a shortcut key can already use `alt+A` accelerator key to press "Add" button and the focus even remains in the gestures tree.
